src/agoradatatools/great_expectations/gx/plugins/expectations/expect_column_nested_field_values_to_be_mostly_not_null.py

```diff
@@ -0,0 +1,329 @@
+"Custom expectation rule that counts the percentage of nulls in a targeted field"
+
+from typing import Dict, Optional, List, Any
+import pandas as pd
+import json
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import (
+    ExecutionEngine,
+    PandasExecutionEngine,
+)
+from great_expectations.expectations.expectation import (
+    ColumnAggregateExpectation,
+    InvalidExpectationConfigurationError,
+)
+from great_expectations.expectations.metrics import (
+    ColumnAggregateMetricProvider,
+    column_aggregate_value,
+)
+
+METRIC_NAME = "column.nested_object_not_null_ratio"
+
+# This method implements the core logic for the PandasExecutionEngine
+class ColumnNestedObjectNotNull(ColumnAggregateMetricProvider):
+    """A custom Great Expectations metric that calculates the proportion of non-null values
+    for a specified key across nested lists of dictionaries in a column."""
+
+    metric_name = METRIC_NAME
+    value_keys = ("non_null_threshold", "target_field")
+
+    @staticmethod
+    def safe_parse(value: str) -> List[Dict[str, Any]]:
+        """
+        Load a JSON string and return a list of dictionaries.
+        If the input is not a valid JSON, return an empty list.
+
+        Parameters:
+            value[str]: the json string to be parsed. If input is not a valid json string, return an empty list
+
+        Returns:
+            Parsed json object or fall back to an empty list
+        """
+        try:
+            parsed = json.loads(value)
+            if isinstance(parsed, list):
+                return parsed
+            else:
+                return []
+        # Fallback if it's not valid JSON like "null"
+        except (json.JSONDecodeError, TypeError):
+            return []
+
+    @column_aggregate_value(engine=PandasExecutionEngine)
+    def _pandas(cls, column: pd.Series, **kwargs) -> float:
+        """
+        Computes the proportion of non-null values for a specified field within a
+        nested list of dictionaries contained in a Pandas column.
+
+        The function flattens each cell's nested structure and checks whether the
+        specified `target_field` exists and is not None.
+
+        Parameters:
+            column (pd.Series): A column where each row contains a list of dictionaries.
+            **kwargs:
+                target_field (str):  the field to validate within each JSON object.
+                non_null_threshold  (float): non null threshold
+        Raises:
+            ValueError: if no json object to be evaluated
+
+        Returns:
+            float: The proportion of non-null entries for `target_field` across all dictionaries.
+        """
+        target_field = kwargs.get("target_field")
```

Comment:
do you want to add a check to ensure `target_field` is not None before passing it onto the logic below?

Reply:
I see - thanks!